### PR TITLE
feat: add setting for scan mode

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Sometimes, automatic scanning is not wanted. To be able to decide whether to use automatic scanning (on start-up & save) or manual scanning via explicit mouse-click, this PR adds a setting to the settings catalogue.

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

image

### CODE REVIEW

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/PreferencesPage.java [hunk 1]
```diff
@@ -18,114 +19,119 @@
...
+        snykCodeSecurityCheckbox = new BooleanFieldEditor(Preferences.ACTIVATE_SNYK_CODE, "Snyk Code enabled",
+            getFieldEditorParent());
+
+        addField(snykCodeSecurityCheckbox);
+
+        addField(new BooleanFieldEditor(Preferences.ACTIVATE_SNYK_IAC, "Snyk Infrastructure-as-Code enabled",
+            getFieldEditorParent()));
+
+        addField(space());
+        addField(new BooleanFieldEditor(Preferences.SCANNING_MODE_AUTOMATIC, "Scan automatically on start-up and save", getFieldEditorParent()));
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/PreferencesPage.java:61]: This is the new checkbox!

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/PreferencesPage.java [hunk 2]
```diff
@@ -18,114 +19,119 @@
...
+        snykView.disableRunAbortActions();
+        snykView.toggleRunActionEnablement();
+        disableSnykCodeIfOrgDisabled();
+
+        new LsConfigurationUpdater().configurationChanged();
+        return superOK;
+    }
+
+    private void disableSnykCodeIfOrgDisabled() {
+        Display.getCurrent().asyncExec(new Runnable() {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/PreferencesPage.java:115]: Added this to not block the UI thread.

File: mvnw.cmd
Comment (bastiandoetsch) [inline mvnw.cmd:1]: No need to review this file - automatically generated.

**Review by cat2608 — APPROVED** [review]
Awesome! 🚀 🙌

**Pull request merged**